### ISSUE TITLE
Issues between CSS file names and their reference in the production js

### ISSUE THREAD

**Issue by sonikdesign** [laynezh/vite-plugin-lib-assets#64]
I put my code for example. I omitted content of some files because they could be any. App.vue for example is just a demo view of the methods defined in main.ts. 

The goal of my app is to create a library to dynamically call assets (images) and CSS styles based on the "client",  "file" or "sheet" parameter. This library, published and executed within another app, only works if I modify node_modules/APP-NAME/assets.js, correcting the two methods that call the CSS files. Methods that retrieve images work correctly.

**package.json**
```json
{
  "name": "example",
  "type": "module",
  "module": "dist/assets.js",
  "types": "dist/main.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "vite",
    "build": "vue-tsc && vite build",
    "preview": "vite preview",
  },
  "dependencies": {
    "vue": "^3.3.11"
  },
  "peerDependencies": {
    "vue": "^3.0.0"
  },
  "devDependencies": {
    "@laynezh/vite-plugin-lib-assets": "^0.5.15",
    "@vitejs/plugin-vue": "^4.5.2",
    "sass": "^1.54.9",
    "typescript": "^5.2.2",
    "vite": "^5.0.8",
    "vite-plugin-dts": "^3.7.2",
    "vue-tsc": "^1.8.25"
  }
}
```

**vite.config.ts**
```ts
import { defineConfig } from 'vite'
import { resolve } from 'path'
import vue from '@vitejs/plugin-vue'
import dts from 'vite-plugin-dts'
import libAssetsPlugin from '@laynezh/vite-plugin-lib-assets'

// https://vitejs.dev/config/
export default defineConfig({
  root: './',
  plugins: [
    vue(),
    dts(),
    libAssetsPlugin({ /* options */ })
  ],
  resolve: {
    alias: {
      $fonts: resolve('./src/assets/common/fonts'),
    }
  },
  build: {
    lib: {
      entry: resolve(__dirname, 'src/main.ts'),
      name: 'assets',
      fileName: 'assets',
      formats: ['es'],
    },
    // assetsInlineLimit: 0,
    copyPublicDir: false,
    cssCodeSplit: true,
    rollupOptions: {
      output: {
        assetFileNames: (assetInfo) => {
          let extType = assetInfo.name.split('.').at(1);
          if (/png|jpe?g|svg|gif|tiff|bmp|ico/i.test(extType)) {
            extType = 'img';
          }
          if (/scss|css|sass/i.test(extType)) {
            extType = 'css';
          }
          return `assets/${extType}/[name]-[hash][extname]`;
        },
        // chunkFileNames: 'assets/js/[name]-[hash].js',
        // entryFileNames: '[name].js',
        // Provide global variables to use in the UMD build
        // for externalized deps
        globals: {
          vue: 'Vue',
        },
      },
      // make sure to externalize deps that shouldn't be bundled
      // into your library
      external: ['vue'],
    },
  }
})
```
**tsconfig.json**
```json
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "skipLibCheck": true,

    /* Bundler mode */
    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "preserve",

    /* Linting */
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["src"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

**tsconfig.node.json**
```json
{
  "compilerOptions": {
    "composite": true,
    "skipLibCheck": true,
    "module": "ESNext",
    "moduleResolution": "bundler",
    "allowSyntheticDefaultImports": true
  },
  "include": ["vite.config.ts"]
}
```

**index.html**
```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Example</title>
  </head>
  <body>
    <div id="app"></div>
    <script type="module" src="/src/app.ts"></script>
  </body>
</html>
```

**public**
```
vite.svg
```

**src/assets**
```
common
-- fonts/*.ttf
-- img/*.png, *.jpg
-- styles/common-style.scss

clientX
-- img/*.png, *.jpg
-- styles/custom-clientX.scss
```

**src/app.ts**
```ts
import { createApp } from 'vue'
import './style.css'
import App from './App.vue'

createApp(App).mount('#app')

```

**src/style.css**
```
Style for demo page
```

**src/App.vue**
```
Demo page
```

**src/main.ts**
```ts
// Img
function getImgUrl(file: string): string {
  return new URL(`./assets/common/img/${file}`, import.meta.url).href
}

function getClientImgUrl(client: string, file: string): string {
  return new URL(`./assets/${client}/img/${file}`, import.meta.url).href
}

// Style
function importStyle(sheet: string) {
  // common- prefix per test purposes. 'sheet' will be 'style'
  return import(`./assets/common/styles/common-${sheet}.scss`)
}

function importClientStyle(client: string) {
  // custom- prefix per test purposes. 'client' will be 'ClientX'
  return import(`./assets/${client}/styles/custom-${client}.scss`)
}

export {
  getImgUrl,
  getClientImgUrl,
  importStyle,
  importClientStyle
}
```

---

This is the dist:

```
assets/*.png, *.jpg, *.ttf
---
00c6a303ec367110.png
01c852359543683a.ttf
7e3efbd427005a95.jpg
...

assets/css/*.css
---
common-style-HhyJmhc_.css
custom-clientX-OOyN9GMZ.css

app.d.ts
---
Empty

main.d.ts
---
declare function getImgUrl(file: string): string;
declare function getClientImgUrl(client: string, file: string): string;
declare function importStyle(sheet: string): Promise<any>;
declare function importClientStyle(client: string, sheet: string): Promise<any>;
export { getImgUrl, getClientImgUrl, importStyle, importClientStyle };

assets.js
---
import t from "./assets/7e3efbd427005a95.jpg";
import a from "./assets/00c6a303ec367110.png";
const n = (s, e) => {
  const o = s[e];
  return o ? typeof o == "function" ? o() : Promise.resolve(o) : new Promise((_, r) => {
    (typeof queueMicrotask == "function" ? queueMicrotask : setTimeout)(r.bind(null, new Error("Unknown variable dynamic import: " + e)));
  });
};
function d(s) {
  return new URL((/* @__PURE__ */ Object.assign({ "./assets/common/img/example1.jpg": t, "./assets/common/img/example2.png": a }))[`./assets/common/img/${s}`], import.meta.url).href;
}
function w(s, e) {
  return new URL((/* @__PURE__ */ Object.assign({ "./assets/common/img/example1.jpg": t, "./assets/common/img/example2.png": a}))[`./assets/${s}/img/${e}`], import.meta.url).href;
}
function $(s) {
  return n(/* @__PURE__ */ Object.assign({ "./assets/common/styles/common-style.scss": () => import("./common-style-4OAPtm6b.js") }), `./assets/common/styles/common-${s}.scss`);
}
function U(s, e) {
  return n(/* @__PURE__ */ Object.assign({ "./assets/clientX/styles/custom-clientX.scss": () => import("./custom-clientX-4HzgcEVU.js")}), `./assets/${s}/styles/custom-${e}.scss`);
}
export {
  w as getClientImgUrl,
  d as getImgUrl,
  U as importClientStyle,
  $ as importStyle
};
```

---

With or without rollupOptions assetFileNames custom function, there's a missmatch between the css files and their import urls in build js functions.
I don't understand if I need to add some definitions to your library options or if it's a Vite problem during build.

I hope I haven't forgotten anything and have made myself understood. Thanks for the support.

**Comment by sonikdesign** [comment]
It seems to work for now by defining in `vite.config.ts` the `chunkFileNames` parameter as follows:

```ts
build: {
rollupOptions: {
  output: {
    chunkFileNames: 'assets/[name].css',
    ...
```

**Comment by coder-layne** [comment]
@sonikdesign It's a bit complicated. Could you share a reproducible version on CodeSandbox or a GitHub repository?

**Comment by sonikdesign** [comment]
@coder-layne my problem was solved by Vite@5.1.0-beta.6. Your vite-plugin-lib-assets does the rest of the job ;-)

ASAP I will publish a repository of my small project for sharing assets and CSS styles with your plugin.
